### PR TITLE
Add moddable distillation dir

### PR DESCRIPTION
**Pull request by jinPrelude**
Add new configs "distillation_path_dir" so you can specify the path where the distillation data will be stored.

### CODE REVIEW

File: rl_algorithms/common/buffer/distillation_buffer.py [hunk 1]
```diff
@@ -13,13 +13,13 @@
...
     """Class for managing reading and writing of distillation data.
-       Distillation data is stored in the buffer_path location on the actual hard disk.
+       Distillation data is stored in the distillation_dataset_path location on the actual hard disk.
        The data collected by the teacher is stored as individual pickle files.
        It is also read as batch through the pytorch DataLoader class.
 
     Attributes:
         batch_size (int): size of batch size from distillation buffer for training
-        buffer_path (list): list of distillation buffer path
+        distillation_dataset_path (list): list of distillation buffer path
```
Comment (MrSyee) [inline rl_algorithms/common/buffer/distillation_buffer.py:22]: I think `distillation_dataset_path` can be changed `dataset_path` due to it is in `DistillationBuffer`

File: configs/pong_no_frameskip_v4/distillation_dqn.py [hunk 2]
```diff
@@ -32,12 +32,13 @@
             "backbone.cnn.cnn_1.cnn",
             "backbone.cnn.cnn_2.cnn",
         ],
-        buffer_path=[
+        # Distillation
+        distillation_dataset_path=[
```
Comment (MrSyee) [inline configs/pong_no_frameskip_v4/distillation_dqn.py:36]: `distillation_path_dir` + `distillation_dataset_path`

**Comment by MrSyee** [comment]
slack test

**Review by MrSyee — APPROVED** [review]

File: rl_algorithms/distillation/dqn_agent.py [hunk 3]
```diff
@@ -46,18 +46,21 @@ def _initialize(self):
...
-                self.hyper_params.batch_size, self.buffer_path, self.log_cfg.curr_time,
+                self.hyper_params.batch_size, self.dataset_path, self.log_cfg.curr_time,
             )
             if self.args.test:
                 self.make_distillation_dir()
 
     def make_distillation_dir(self):
         """Make directory for saving distillation data."""
+        if self.hyper_params.save_dir[-1] != "/":
+            self.hyper_params.save_dir += "/"
```
Comment (jinPrelude) [inline rl_algorithms/distillation/dqn_agent.py:60]: replace this code by using os.join

Comment (jinPrelude) [reply rl_algorithms/distillation/dqn_agent.py]: resolved in 2ea39fe

File: rl_algorithms/common/buffer/distillation_buffer.py (hunk 1 above)
Comment (jinPrelude) [reply rl_algorithms/common/buffer/distillation_buffer.py]: resolved in 661a128

**Pull request merged**